src/client/pages/OfferNew/Introduction/Sidebar/CancellationOptions.tsx

```diff
@@ -70,78 +61,80 @@ export const CancellationOptions: React.FC<CancellationOptionsProps> = ({
   )
 }
 
-interface QuoteCancellationOptionProps {
-  isGenericQuote: boolean
+const getLabelContent = (textKeys: TextKeyMap, quote: OfferQuote) => {
+  if (quote.currentInsurer?.displayName) {
+    return textKeys.SIDEBAR_REQUEST_CANCELLATION_INSURANCE_NAME_PROVIDER_LABEL({
+      INSURANCE_NAME: quote.displayName,
+      INSURANCE_PROVIDER: quote.currentInsurer.displayName,
+    })
+  }
+
+  return textKeys.SIDEBAR_REQUEST_CANCELLATION_INSURANCE_NAME_LABEL({
+    INSURANCE_NAME: quote.displayName,
+  })
+}
+
+type QuoteCancellationOptionProps = {
   quote: OfferQuote
   setShowError: (showError: boolean) => void
   refetch: () => Promise<void>
   handleFail: (e: Error) => void
 }
+
 const QuoteCancellationOption: React.FC<QuoteCancellationOptionProps> = ({
-  isGenericQuote,
   quote,
   setShowError,
   handleFail,
   refetch,
 }) => {
   const textKeys = useTextKeys()
+
   const [isLoading, setIsLoading] = React.useState(false)
+
   const [setStartDate] = useStartDateMutation()
   const [removeStartDate] = useRemoveStartDateMutation()
 
   const checked = !quote.startDate
+  const checkboxLabel = getLabelContent(textKeys, quote)
+
+  const toggle = async () => {
+    try {
+      setShowError(false)
+      setIsLoading(true)
 
-  const toggle = () => {
-    setShowError(false)
-    setIsLoading(true)
-    if (!checked) {
-      removeStartDate({
-        variables: {
-          quoteId: quote.id,
-        },
-      })
-        .then(() => refetch())
-        .catch(handleFail)
-        .finally(() => setIsLoading(false))
-    } else {
-      setStartDate({
-        variables: {
-          quoteId: quote.id,
-          date: format(new Date(), gqlDateFormat),
-        },
-      })
-        .then(() => refetch())
-        .catch(handleFail)
-        .finally(() => setIsLoading(false))
+      if (!checked) {
+        await removeStartDate({
+          variables: {
+            quoteId: quote.id,
+          },
+        })
+        await refetch()
```

Comment:
Question: Is this `await` necessary when this is the last thing that happens in this block?

Reply:
Yes. That's because checkbox state (checked/unchecked) is defined based API data. So here we're executing some mutations when user clicks on the checkbox and then we need to wait until we get new data changed by the mutations before removing the loading state and show the checkbox with the correct state. This is the result in case I remove it:

https://user-images.githubusercontent.com/19200662/152811783-85229ba6-4fe7-4b02-86a4-7d5189616385.mov

Reply:
Ah yes 👍